analysis/03_new_model_training/06_grid_To_mun.md

```diff
@@ -0,0 +1,363 @@
+df_weight.head()
+```
+
+### Following Steps are to convert grid_based model into Municipality based one
+
+```python
+# Join df and df_weight
+join_df = df.merge(df_weight, on="grid_point_id", how="left")
+join_df.head()
+```
+
+```python
+# Hist plot after data stratification
+bins2 = [0, 0.00009, 1, 10, 50, 101]
+samples_per_bin2, binsP2 = np.histogram(
+    join_df["percent_buildings_damaged"], bins=bins2
+)
+plt.figure(figsize=(4, 3))
+plt.xlabel("Damage Values")
+plt.ylabel("Frequency")
+plt.plot(binsP2[1:], samples_per_bin2)
+```
+
+```python
+# Check the bins' intervalls
+join_df["percent_buildings_damaged"].value_counts(bins=binsP2)
+```
+
+```python
+# Remove zeros from wind_speed
+join_df = join_df[(join_df[["wind_speed"]] != 0).any(axis=1)]
+join_df = join_df.drop(columns=["grid_point_id", "typhoon_year"])
+join_df.head()
+```
+
+```python
+# Hist plot after removing rows where windspeed is 0
+bins2 = [0, 0.00009, 1, 10, 50, 101]
+samples_per_bin2, binsP2 = np.histogram(
+    join_df["percent_buildings_damaged"], bins=bins2
+)
+plt.figure(figsize=(4, 3))
+plt.xlabel("Damage Values")
+plt.ylabel("Frequency")
+plt.plot(binsP2[1:], samples_per_bin2)
+```
+
+```python
+print(samples_per_bin2)
+print(binsP2)
+```
+
+```python
+# Check the bins' intervalls
+join_df["percent_buildings_damaged"].value_counts(bins=binsP2)
+```
+
+```python
+bin_index2 = np.digitize(join_df["percent_buildings_damaged"], bins=binsP2)
+```
+
+```python
+y_input_strat = bin_index2
+```
+
+```python
+features = [
+    "wind_speed",
+    # "track_distance",
+    # "total_buildings",
+]
+
+# Split X and y from dataframe features
+X = join_df[features]
+display(X.columns)
+y = join_df["percent_buildings_damaged"]
+
+scaler = preprocessing.StandardScaler().fit(X)
+X_scaled = scaler.transform(X)
+```
+
+```python
+# Define train-test-split function
+
+X_train, X_test, y_train, y_test = train_test_split(
+    X_scaled,
+    join_df["percent_buildings_damaged"],
+    stratify=y_input_strat,
+    test_size=0.2,
+)
+
+# Define XGBoost Reduced Overfitting model
+xgb = XGBRegressor(
+    base_score=0.5,
+    booster="gbtree",
+    colsample_bylevel=0.8,
+    colsample_bynode=0.8,
+    colsample_bytree=0.8,
+    gamma=3,
+    eta=0.01,
+    importance_type="gain",
+    learning_rate=0.1,
+    max_delta_step=0,
+    max_depth=4,
+    min_child_weight=1,
+    missing=1,
+    n_estimators=100,
+    early_stopping_rounds=10,
+    n_jobs=1,
+    nthread=None,
+    objective="reg:squarederror",
+    reg_alpha=0,
+    reg_lambda=1,
+    scale_pos_weight=1,
+    seed=None,
+    silent=None,
+    subsample=0.8,
+    verbosity=1,
+    eval_metric=["rmse", "logloss"],
+    random_state=0,
+)
+
+eval_set = [(X_test, y_test)]
+xgb_model = xgb.fit(X_train, y_train, eval_set=eval_set, verbose=False)
+
+X2 = sm.add_constant(X_train)
+est = sm.OLS(y_train, X2)
+est2 = est.fit()
+print(est2.summary())
+```
+
+```python
+# Combine test and training set to have predictiob for all
```

Comment:
I think in this case there's not much point in separating the dataset, maybe you can just use all the data for training